test/middleware/retry_test.rb

```diff
@@ -195,5 +195,22 @@ def test_should_call_retry_if_for_empty_method_list
       assert_equal 2, @times_called
     end
 
+    def test_should_rewind_files_on_retry
+      io = StringIO.new("Test data")
+      upload_io = Faraday::UploadIO.new(io, "application/octet/stream")
+
+      rewound = 0
+      rewind = lambda { rewound += 1 }
+
+      upload_io.stub :rewind, rewind do
+        @explode = lambda {|n| raise Errno::ETIMEDOUT }
+        check = lambda { |env,exception| true }
+        assert_raises(Errno::ETIMEDOUT) {
+          conn(:retry_if => check).post("/unstable", { file: upload_io})
```

Comment:
_Trivial_ note: Can you add a space to the end of this Hash `{ file: upload_io}`?

Other stylistic note: All other Hashes use hashrockets, perhaps this one should, too?